xblock/plugin.py

```diff
@@ -41,6 +41,32 @@ def default_select(identifier, all_entry_points):  # pylint: disable=inconsisten
         raise AmbiguousPluginError(all_entry_points)
 
 
+def select_with_overrides(identifier, all_entry_points):
```

Comment:
I think we should keep this method private to avoid its availability to library users

Reply:
+1

Reply:
I like your instinct @farhan to avoid expanding the public API unnecessarily. Another approach would be call this new function `default_select`, since it is indeed the default selection method now. The old default select could become a private method, like `_default_select_no_override`

Reply:
or we can remove `_default_select_no_override` as its not required anymore for simplicity.